src/main/java/com/zendesk/maxwell/MaxwellHA.java

```diff
@@ -83,4 +109,51 @@ public void startHA() throws Exception {
 
 		Thread.sleep(Long.MAX_VALUE);
 	}
+
+	/**
+	 * indicates that Ha is started in zookeeper mode
+	 * @throws Exception
+	 */
+	public void startHAZookeeper() throws Exception {
+		String electPath = "/" + clientID + "/services";
+		String masterPath = "/" + clientID + "/leader";
+		CuratorUtils cu = new CuratorUtils();
+		cu.setZookeeperServer(zookeeperServer);
+		cu.setSessionTimeoutMs(sessionTimeoutMs);
+		cu.setConnectionTimeoutMs(connectionTimeoutMs);
+		cu.setMaxRetries(maxRetries);
+		cu.setBaseSleepTimeMs(baseSleepTimeMs);
+		cu.setClientId(clientID);
+		cu.setElectPath(electPath);
+		cu.setMasterPath(masterPath);
+		cu.init();
+		CuratorFramework client = cu.getClient();
+		LeaderLatch leader = new LeaderLatch(client, cu.getElectPath());
+		leader.start();
+		LOGGER.info("this node is participating in the election of the leader ....");
+		leader.addListener(new LeaderLatchListener() {
+			@Override
+			public void isLeader() {
+				try {
+					cu.register();
+				} catch (Exception e) {
+					e.printStackTrace();
+					LOGGER.error("The node registration is abnormal, check whether the maxwell host communicates properly with the zookeeper network");
+					cu.stop();
+					System.exit(1);
+				}
+				LOGGER.info("node is current leader, starting Maxwell....");
+				run();
+				cu.stop();
+			}
+
+			@Override
+			public void notLeader() {
+				//LeaderLatch.CloseMode.SILENT mode will not invoke this method
```

Comment:
Are you expecting this to never happen or what makes it OK for us to ignore the loss of leadership here?